subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyConstraintHandler.java

```diff
@@ -68,7 +69,7 @@ public DependencyConstraint create(Object dependencyNotation, Action<? super Dep
     @Override
     public DependencyConstraint platform(Object notation) {
         DependencyConstraint dependencyConstraint = create(notation);
-        PlatformSupport.addPlatformAttribute(dependencyConstraint, PlatformSupport.REGULAR_PLATFORM);
+        PlatformSupport.addPlatformAttribute(dependencyConstraint, toCategory(Category.REGULAR_PLATFORM));
```

Comment:
:o: We could probably pre-compute `enforced platform` and `regular platform` in the constructor, to return the same instance for all dependencies.